Tags to sections

pixelchef opened the pull request.

pixelchef committed:
Update blog_data.rb

pixelchef committed:
Update tag_pages.rb

pixelchef committed:
Update tag_pages.rb

pixelchef committed:
Create section_pages.rb

pixelchef committed:
Update tag_pages.rb

pixelchef committed:
Update extension.rb

pixelchef committed:
Update blog_data.rb

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -63,7 +63,7 @@ def tags
 
         @_articles.each do |article|
-          article.tags.each do |tag|
+          article.sections[section.name].each do |tag|
```
article.section.each

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -63,7 +63,7 @@ def tags
         @_articles.each do |article|
-          article.tags.each do |tag|
+          article.sections[section.name].each do |tag|
             tags[tag] ||= []
```
tags[section.name]

mrh-is commented on lib/middleman-blog/section_pages.rb:
```diff
@@ -0,0 +1,53 @@
+        @tag_template = blog_controller.options.tag_template
+        @blog_data = blog_controller.data
+
+        @generate_tag_pages = blog_controller.options.generate_tag_pages
```
generate_section_pages

mrh-is commented on lib/middleman-blog/section_pages.rb:
```diff
@@ -0,0 +1,53 @@
+      def initialize(app, blog_controller)
+        @sitemap = app.sitemap
+        @blog_controller = blog_controller
+        @tag_link_template = uri_template blog_controller.options.taglink
```
Need new option for this

pixelchef replied:
Im not sure where to add options?

pixelchef replied:
I think I did it!

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -63,7 +63,7 @@ def tags
         tags = {}
```
This function not renamed sections

pixelchef committed:
Update blog_data.rb

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
       # of BlogArticles associated with that tag.
       # @return [Hash<String, Array<Middleman::Sitemap::Resource>>]
       def tags
```
Rename sections

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
       # @return [Hash<String, Array<Middleman::Sitemap::Resource>>]
       def tags
-        tags = {}
+        section = {}
```
sections

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
         @_articles.each do |article|
-          article.tags.each do |tag|
+          article.section.each.each do |tag|
             tags[tag] ||= []
```
sections[section.name] on both lines

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
         end
 
         # Sort each tag's list of articles
         tags.each do |tag, articles|
```
sections.each do |section, articles|

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
         # Sort each tag's list of articles
         tags.each do |tag, articles|
-          tags[tag] = articles.sort_by(&:date).reverse
+          tags[section.name] = articles.sort_by(&:date).reverse
```
sections[section]

mrh-is commented on lib/middleman-blog/blog_data.rb:
```diff
@@ -60,18 +60,18 @@ def local_articles(locale=::I18n.locale)
+          tags[section.name] = articles.sort_by(&:date).reverse
         end
 
         tags
```
sections

pixelchef committed:
option added

pixelchef committed:
section template

pixelchef committed:
Update section_pages.rb

pixelchef committed:
Update blog_data.rb

pixelchef committed:
Update blog_data.rb

pixelchef committed:
Update blog_data.rb

pixelchef committed:
Update blog_data.rb

pixelchef commented:
I gave you editing right cause im like 
![slack-imgs](https://cloud.githubusercontent.com/assets/4326753/22811641/ec681182-eef4-11e6-8be7-1583fc61e1c6.gif)

mrh-is committed:
Update blog_data.rb

mrh-is committed:
Update extension.rb

mrh-is committed:
Update section_pages.rb

mrh-is committed:
Update blog_data.rb

pixelchef commented:
Oddly enough the options are breaking the build despite being defined in config.rb Why com you no work easy mm

```RuntimeError: Setting sectionlink doesn't exist
  /usr/local/lib/ruby/gems/2.3.0/gems/middleman-core-4.1.8/lib/middleman-core/configuration.rb:75:in `define_setting'
  /usr/local/lib/ruby/gems/2.3.0/gems/middleman-core-4.1.8/lib/middleman-core/configuration.rb:37:in `[]='
  /usr/local/lib/ruby/gems/2.3.0/gems/middleman-core-4.1.8/lib/middleman-core/configuration.rb:46:in `method_```

mrh-is commented:
Does it give you an error message? [These lines](https://github.com/middleman/middleman/blob/master/middleman-core/lib/middleman-core/configuration.rb#L75-L77) indicate that it should give you some error message. Which one is probably useful (though I've never been in this part of the code, so I dunno how helpful it would be).

pixelchef commented:
No error message like that. Just output saying the key doesn't exists [when it does](https://github.com/pixelchef/middleman-blog/pull/1/files#diff-1219fd2a8ef0bc899ac4bdab3e8086c5R19). I've run bundle install again thinking it wasn't pulling the latest with no success.

The syntax looks right here as well https://github.com/pixelchef/middleman-blog/pull/1/files#diff-675ae966f581d0f79cdd1defb2def3e7R13

Also assuming I've added them as options to the blog correctly here in config.rb 

```  
blog.section_template = "#{country_code}/news/section.html"
blog.sectionlink = "#{country_code unless country_code == :en}/news/sections/{section}.html"
```

and here in my Gemfile
```
gem 'middleman-blog', :git => "https://github.com/pixelchef/middleman-blog.git", :branch => "pixelchef-patch-1"
```

The site is loading now but isn't showing the section pages. It is showing tag pages. I've added the template next to the tags page (so it has the page path per the config file below. I'm guessing it's just not finding any of the `section.name` items in the article's frontmatter?

pixelchef commented:
The syntax looks right here as well https://github.com/pixelchef/middleman-blog/pull/1/files#diff-675ae966f581d0f79cdd1defb2def3e7R13

mrh-is commented:
Based on the console output you had before, it looks like [this line](https://github.com/middleman/middleman/blob/master/middleman-core/lib/middleman-core/configuration.rb#L75) is raising the exception, which means that the config was finalized somehow. I poked around a bit and didn't see anything obvious about how that was getting done.

mrh-is commented:
This might all be way over engineered, it looks like [custom collections](https://middlemanapp.com/basics/blogging/#custom-article-collections) might do what you want here.

EDIT: You might have to poke around [in here](https://github.com/middleman/middleman-blog/blob/master/lib/middleman-blog/custom_pages.rb) to use `section.name` instead of `section` (which might work by itself, I dunno). Also not sure how it will work with an array, it doesn't look like this code knows about array properties. But still, this feature does 90% of what you're looking for AFAICT.

pixelchef commented:
True I could put something like this and duplicate the `tags.erb`. I don't know how to solve the array issue :(

```
    blog.custom_collections = {
      category: {
        template: "#{country_code}/news/section.html",
        link: "#{country_code unless country_code == :en}/news/sections/#{section.name}.html"
      }
    }
```

mrh-is commented:
Methinks you'd monkey patch like this (wrote all this in Github, so bugs probably, but I added comments):

```ruby
module Middleman
  module Blog
    class CustomPages
      def initialize(property, app, controller, options)
        @property = property
        @sitemap = app.sitemap
        @blog_controller = controller
        @blog_data = controller.data
        @link_template = uri_template options[:link]
        @page_template = options[:template]
        # New "transform" property! Lets you define how to get the value out
        # You get the article and the property you're looking for, and you return the value you care about
        # Defaults to just looking up the name in the front matter, as before (so this wouldn't break the previous behavior)
        # (A Proc is like a JS callback/closure/block)
        @transform = options[:transform] || Proc.new { |article, property| article.data[property] }
      end

      def manipulate_resource_list(resources)
        # Get the articles by property in the helper below, the rest of this is the same
        resources + articles_by_property.map do |property_value, articles|
          build_resource(link(property_value), property_value, articles)
        end
      end

      def articles_by_property
        # This is just a copy from the BlogData.tags method, with new names for things
        articles_by_property = {}

        @blog_data.articles.each do |article|
          # Use the transform method to get the property value
          # Use `Array` to make sure it's an array so we can use `each` (makes this work for single values or multiple values [like tags])
          Array(transform(article, property)).each do |property_value|
            articles_by_property[property_value] ||= []
            articles_by_property[property_value] << article
          end
        end

        # Don't have to sort, `build_resource` already does that

        articles_by_property
      end
    end
  end
end
```

So that adds a `transform` option to the hash, which you use like this:

```ruby
    blog.custom_collections = {
      section: {
        template: "/news/section.html",
        link: "#{country_code unless country_code == :en}/news/sections/{section}.html",
        # We're ignoring the `property`, since we already know what it is
        transform: Proc.new { |article, property| article.data["section"]["name"] }
      }
    }
```

pixelchef commented:
Won't that link fail as section is an array in the custom collections?

mrh-is commented:
I don't see any obvious failures... but then again I'm not running this code. 😛 The code I added above lets one article be in multiple sections, but `articles_by_property` is the same type as before (`[Hash<String, Array<Middleman::Sitemap::Resource>>]`). So it should just work.

pixelchef commented:
Hmm just ran it with this error where 'link' in the custom_collections fails 

```bundler: failed to load command: middleman (/usr/local/bin/middleman)
NameError: undefined local variable or method `section' for #<Middleman::ConfigContext:0x007f9ab5a3ce18>
  /Users/patrickrogers/Documents/realm.io-middleman/config.rb:180:in `block (2 levels) in evaluate_configuration!'
  /usr/local/lib/ruby/gems/2.3.0/gems/middleman-core-4.1.8/lib/middleman-core/extension.rb:387:in `setup_options'
  /usr/local/lib/ruby/gems/2.3.0/gems/middleman-core-4.1.8/lib/middleman-core/extension.rb:314:in `initialize'
  /usr/local/lib/ruby/gems/2.3.0/bundler/gems/middleman-blog-e38ee99f0fb9/lib/middleman-blog/extension.rb:67:in `initialize'
```

mrh-is commented:
What's on `config.rb:180`?

pixelchef commented:
this error where 'link' in the custom_collections fails

mrh-is commented:
Oh oops, I had an error in there, I updated that line above, try again! (`#{section}` => `{section}` [no `#`])

pixelchef commented:
Ah. Now all assets are trying to build :) 

```
$ bundle exec middleman s
^[[A== The Middleman is loading
== Locales: en, cn, kr, jp (Default en)
== Blog Sources: data/library/en/{id}.html (:prefix + :sources)
== Blog Sources: data/library/cn/{id}.html (:prefix + :sources)
== Blog Sources: data/library/jp/{id}.html (:prefix + :sources)
== Blog Sources: data/library/kr/{id}.html (:prefix + :sources)
bundler: failed to load command: middleman (/usr/local/bin/middleman)
RuntimeError: Path cn/news/sections/middleman-util-enhancedhash-id-7a1bguxooquq2wuquu8euo-name-other.html proxies to unknown file cn/news/section.html:["assets/benchmarks/benchmarks/benchmarks.001.png", "assets/benchmarks/benchmarks/benchmarks.001b.png", "assets/benchmarks/benchmarks/benchmarks.002.png", "assets/benchmarks/benchmarks/benchmarks.002b.png", "assets/benchmarks/benchmarks/benchmarks.003.png", "assets/benchmarks/benchmarks/benchmarks.003b.png", "assets/benchmarks/benchmarks-android/benchmarks-android.001.png", "assets/benchmarks/benchmarks-android/benchmarks-android.002.png", "assets/benchmarks/benchmarks-android/benchmarks-android.003.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.001.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.002.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.003.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.004.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.005.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.006.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.001.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.002.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.003.png", "assets/css/global/site-china.css", "assets/css/global/site-japan.css", "assets/css/global/site-korea.css", "assets/css/pages/about.css", "assets/css/pages/addons.css", "assets/css/pages/advent.css", "assets/css/pages/conf.css", "assets/css/pages/contact.css", "assets/css/pages/content-library.css", "assets/css/pages/docs.css", "assets/css/pages/home.css", "assets/css/pages/jobs.css", "assets/css/pages/languages/languages.css", "assets/css/pages/legal.css", "assets/css/pages/news.css", "assets/css/pages/ns-predicate.css", "assets/css/pages/post-truncate.css", "assets/css/pages/post.css", "assets/css/pages/press.css", "assets/css/pages/price.css", "assets/css/pages/product", "assets/css/pages/products/realm-mobile-database.css", "assets/css/pages/products/realm-mobile-platform.css", "assets/css/pages/rmpdocs.css", "assets/css/pages/styleguide.css", "assets/css/pages/users.css", "assets/css/pages/webinar.css", "assets/css/pages/world-tour.css", "assets/css/site.css", "assets/css/vendors/bootstrap.css", "assets/css/vendors/pygments-dark.css", "assets/downloads/NSPredicateCheatsheet.pdf", "assets/downloads/webinars/2016-12-07-realm-offline-first/Realm-Webinar-The-Offline-first-Approach-to-Mobile-App Development.pdf", "assets/downloads/whitepaper/TheOffline-firstApproachtoMobileAppDevelopment-Realm.pdf", "assets/fonts/fontawesome-webfont.eot", "assets/fonts/fontawesome-webfont.svg", "assets/fonts/fontawesome-webfont.ttf", "assets/fonts/fontawesome-webfont.woff", "assets/fonts/fontawesome-webfont.woff2", "assets/fonts/glyphicons-halflings-regular.eot", "assets/fonts/glyphicons-halflings-regular.svg", "assets/fonts/glyphicons-halflings-regular.ttf", "assets/fonts/glyphicons-halflings-regular.woff", "assets/fonts/Realm.woff", "assets/fonts/Realm_Lt.woff", "assets/fonts/Realm_Md.woff", "assets/gifs/altconf-ash-furrow.gif", "assets/gifs/altconf-marin.gif", "assets/gifs/altconf-nathan.gif", "assets/gifs/altconf-spahr.gif", "assets/gifs/fgn.gif", "assets/gifs/finegrainednotifications/add-repo.gif", "assets/gifs/finegrainednotifications/flash-repo.gif", "assets/gifs/finegrainednotifications/initial-repos.gif", "assets/gifs/finegrainednotifications/swap-repos.gif", "assets/gifs/realm-cocoa-deal-with-it.gif", "assets/gifs/slug-3d-touch.gif", "assets/gifs/slug-alexis-demo.gif", "assets/gifs/slug-genady-livephoto.gif", "assets/gifs/slug-kyle-fuller.gif", "assets/gifs/slug-nacho-complexity.gif", "assets/gifs/slug-rachel-demo.gif", "assets/gifs/slug-sash-gameplay.gif", "assets/gifs/slug-sash-mario.gif", "assets/gifs/slug-sash-zats-cat.gif", "assets/gifs/summit-ayaka.gif", "assets/gifs/summit-gem.gif", "assets/gifs/summit-hermes-noescape.gif", "assets/gifs/summit-kyle.gif", "assets/gifs/summit-radek.gif", "assets/gifs/TMI-2.gif", "assets/img/about/hero.png", "assets/img/addons/androidHalfDisplayHeroBack.png", "assets/img/addons/androidHalfDisplayHeroBackMissing.png", "assets/img/addons/androidLanguageIcon.png", "assets/img/addons/iphoneHalfDisplayHero.png", "assets/img/addons/listView.png", "assets/img/addons/mapView.png", "assets/img/addons/mapViewPreview.png", "assets/img/addons/objcLanguageIcon.png", "assets/img/addons/swiftLanguageIcon.png", "assets/img/branding/OS-X-Browser-App-Icon-@2x.png", "assets/img/branding/OS-X-Browser-App-Icon.png", "assets/img/branding/OS-X-Browser-App-Icon.psd", "assets/img/branding/Realm Original Logo CMYK.ai", "assets/img/branding/Realm Original Logo CMYK.eps", "assets/img/branding/Realm Original Logo CMYK.pdf", "assets/img/branding/YouTube-Watermark-A.png", "assets/img/branding/YouTube-Watermark-B.png", "assets/img/conference/altconf/aaronhillegass.jpg", "assets/img/conference/altconf/abizern.jpg", "assets/img/conference/altconf/airspeedswift.jpg", "assets/img/conference/altconf/aleen.jpg", "assets/img/conference/altconf/alexjsinger.jpg", "assets/img/conference/altconf/ashfurrow.jpg", "assets/img/conference/altconf/ashleynh.jpg", "assets/img/conference/altconf/benjaminencz.jpg", "assets/img/conference/altconf/benlodotcom.jpg", "assets/img/conference/altconf/billytobon.jpg", "assets/img/conference/altconf/blizzardblind.jpg", "assets/img/conference/altconf/bmf.jpg", "assets/img/conference/altconf/bricepollock.jpg", "assets/img/conference/altconf/casademora.jpg", "assets/img/conference/altconf/changecatalysts.jpg", "assets/img/conference/altconf/chriseidhof.jpg", "assets/img/conference/altconf/crandycodes.jpg", "assets/img/conference/altconf/crlueck.jpg", "assets/img/conference/altconf/dandbj13.jpg", "assets/i
```

pixelchef commented:
LOL shoot tho

 ```
$ bundle exec middleman s
== The Middleman is loading
== Locales: en, cn, kr, jp (Default en)
== Blog Sources: data/library/en/{id}.html (:prefix + :sources)
== Blog Sources: data/library/cn/{id}.html (:prefix + :sources)
== Blog Sources: data/library/jp/{id}.html (:prefix + :sources)
== Blog Sources: data/library/kr/{id}.html (:prefix + :sources)
bundler: failed to load command: middleman (/usr/local/bin/middleman)
RuntimeError: Path cn/news/sections/middleman-util-enhancedhash-id-7a1bguxooquq2wuquu8euo-name-other.html proxies to unknown file cn/news/section.html:["assets/benchmarks/benchmarks/benchmarks.001.png", "assets/benchmarks/benchmarks/benchmarks.001b.png", "assets/benchmarks/benchmarks/benchmarks.002.png", "assets/benchmarks/benchmarks/benchmarks.002b.png", "assets/benchmarks/benchmarks/benchmarks.003.png", "assets/benchmarks/benchmarks/benchmarks.003b.png", "assets/benchmarks/benchmarks-android/benchmarks-android.001.png", "assets/benchmarks/benchmarks-android/benchmarks-android.002.png", "assets/benchmarks/benchmarks-android/benchmarks-android.003.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.001.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.002.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.003.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.004.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.005.png", "assets/benchmarks/benchmarks-reactnative/benchmarks-reactnative.006.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.001.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.002.png", "assets/benchmarks/benchmarks-xamarin/benchmarks-xamarin.003.png", "assets/css/global/site-china.css", "assets/css/global/site-japan.css", "assets/css/global/site-korea.css", "assets/css/pages/about.css", "assets/css/pages/addons.css", "assets/css/pages/advent.css", "assets/css/pages/conf.css", "assets/css/pages/contact.css", "assets/css/pages/content-library.css", "assets/css/pages/docs.css", "assets/css/pages/home.css", "assets/css/pages/jobs.css", "assets/css/pages/languages/languages.css", "assets/css/pages/legal.css", "assets/css/pages/news.css", "assets/css/pages/ns-predicate.css", "assets/css/pages/post-truncate.css", "assets/css/pages/post.css", "assets/css/pages/press.css", "assets/css/pages/price.css", "assets/css/pages/product", "assets/css/pages/products/realm-mobile-database.css", "assets/css/pages/products/realm-mobile-platform.css", "assets/css/pages/rmpdocs.css", "assets/css/pages/styleguide.css", "assets/css/pages/users.css", "assets/css/pages/webinar.css", "assets/css/pages/world-tour.css", "assets/css/site.css", "assets/css/vendors/bootstrap.css", "assets/css/vendors/pygments-dark.css", "assets/downloads/NSPredicateCheatsheet.pdf", "assets/downloads/webinars/2016-12-07-realm-offline-first/Realm-Webinar-The-Offline-first-Approach-to-Mobile-App Development.pdf", "assets/downloads/whitepaper/TheOffline-firstApproachtoMobileAppDevelopment-Realm.pdf", "assets/fonts/fontawesome-webfont.eot", "assets/fonts/fontawesome-webfont.svg", "assets/fonts/fontawesome-webfont.ttf", "assets/fonts/fontawesome-webfont.woff", "assets/fonts/fontawesome-webfont.woff2", "assets/fonts/glyphicons-halflings-regular.eot", "assets/fonts/glyphicons-halflings-regular.svg", "assets/fonts/glyphicons-halflings-regular.ttf", "assets/fonts/glyphicons-halflings-regular.woff", "assets/fonts/Realm.woff", "assets/fonts/Realm_Lt.woff", "assets/fonts/Realm_Md.woff", "assets/gifs/altconf-ash-furrow.gif", "assets/gifs/altconf-marin.gif", "assets/gifs/altconf-nathan.gif", "assets/gifs/altconf-spahr.gif", "assets/gifs/fgn.gif", "assets/gifs/finegrainednotifications/add-repo.gif", "assets/gifs/finegrainednotifications/flash-repo.gif", "assets/gifs/finegrainednotifications/initial-repos.gif", "assets/gifs/finegrainednotifications/swap-repos.gif", "assets/gifs/realm-cocoa-deal-with-it.gif", "assets/gifs/slug-3d-touch.gif", "assets/gifs/slug-alexis-demo.gif", "assets/gifs/slug-genady-livephoto.gif", "assets/gifs/slug-kyle-fuller.gif", "assets/gifs/slug-nacho-complexity.gif", "assets/gifs/slug-rachel-demo.gif", "assets/gifs/slug-sash-gameplay.gif", "assets/gifs/slug-sash-mario.gif", "assets/gifs/slug-sash-zats-cat.gif", "assets/gifs/summit-ayaka.gif", "assets/gifs/summit-gem.gif", "assets/gifs/summit-hermes-noescape.gif", "assets/gifs/summit-kyle.gif", "assets/gifs/summit-radek.gif", "assets/gifs/TMI-2.gif", "assets/img/about/hero.png", "assets/img/addons/androidHalfDisplayHeroBack.png", "assets/img/addons/androidHalfDisplayHeroBackMissing.png", "assets/img/addons/androidLanguageIcon.png", "assets/img/addons/iphoneHalfDisplayHero.png", "assets/img/addons/listView.png", "assets/img/addons/mapView.png", "assets/img/addons/mapViewPreview.png", "assets/img/addons/objcLanguageIcon.png", "assets/img/addons/swiftLanguageIcon.png", "assets/img/branding/OS-X-Browser-App-Icon-@2x.png", "assets/img/branding/OS-X-Browser-App-Icon.png", "assets/img/branding/OS-X-Browser-App-Icon.psd", "assets/img/branding/Realm Original Logo CMYK.ai", "assets/img/branding/Realm Original Logo CMYK.eps", "assets/img/branding/Realm Original Logo CMYK.pdf", "assets/img/branding/YouTube-Watermark-A.png", "assets/img/branding/YouTube-Watermark-B.png", "assets/img/conference/altconf/aaronhillegass.jpg", "assets/img/conference/altconf/abizern.jpg", "assets/img/conference/altconf/airspeedswift.jpg", "assets/img/conference/altconf/aleen.jpg", "assets/img/conference/altconf/alexjsinger.jpg", "assets/img/conference/altconf/ashfurrow.jpg", "assets/img/conference/altconf/ashleynh.jpg", "assets/img/conference/altconf/benjaminencz.jpg", "assets/img/conference/altconf/benlodotcom.jpg", "assets/img/conference/altconf/billytobon.jpg", "assets/img/conference/altconf/blizzardblind.jpg", "assets/img/conference/altconf/bmf.jpg", "assets/img/conference/altconf/bricepollock.jpg", "assets/img/conference/altconf/casademora.jpg", "assets/img/conference/altconf/changecatalysts.jpg", "assets/img/conference/altconf/chriseidhof.jpg", "assets/img/conference/altconf/crandycodes.jpg", "assets/img/conference/altconf/crlueck.jpg", "assets/img/conference/altconf/dandbj13.jpg", "assets/img/conference/altconf/dazeend.jpg", "assets/img/conference/altconf/dimsumthinking.jpg", "assets/img/conference/altconf/dwiskus.jpg", "assets/img/conference/altconf/fakerapper.jpg", "assets/img/conference/altconf/flargh.jpg", "assets/img/conference/altconf/flexmonkey.jpg", "assets/img/conference/altconf/funkyboy.jpg", "assets/img/conference/altconf/glentiospaulz.jpg", "assets/img/conference/altconf/goawaygeek.jpg", "assets/img/conference/altconf/graiz.jpg", "assets/img/conference/altconf/hodapp.jpg", "assets/img/conference/altconf/holdenkarau.jpg", "assets/img/conference/altconf/icanzilb.jpg", "assets/img/conference/altconf/iwasleeg.jpg", "assets/img/conference/altconf/jaimzuber.jpg", "assets/img/conference/altconf/jamesdempsey.jpg", "assets/img/conference/altconf/jcieplinski.jpg", "assets/img/conference/altconf/jflinter.jpg", "assets/img/conference/altconf/joshjet.jpg", "assets/img/conference/altconf/jshier.jpg", "assets/img/conference/altconf/jspahrsummers.jpg", "assets/img/conference/altconf/kendalldevdiary.jpg", "assets/img/conference/altconf/krotondo.jpg", "assets/img/conference/altconf/kylefuller.jpg", "assets/img/conference/altconf/macsparky.jpg", "assets/img/conference/altconf/mennenia.jpg", "assets/img/conference/altconf/mgglidden.jpg", "assets/img/conference/altconf/mgorbach.jpg", "assets/img/conference/altconf/michaelmace.jpg", "assets/img/conference/altconf/micheletitolo.jpg", "assets/img/conference/altconf/mikecritz.jpg", "assets/img/conference/altconf/milabila.jpg", "assets/img/conference/altconf/mostgood.jpg", "assets/img/conference/altconf/mronge.jpg", "assets/img/conference/altconf/natashatherobot.jpg", "assets/img/conference/altconf/neonacho.jpg", "assets/img/conference/altconf/neror.jpg", "assets/img/conference/altconf/nowallthetime.jpg", "assets/img/conference/altconf/orta.jpg", "assets/img/conference/altconf/robelkin.jpg", "assets/img/conference/altconf/rwenderlich.jpg", "assets/img/conference/altconf/ryannystromacoomans.jpg", "assets/img/conference/altconf/saurik.jpg", "assets/img/conference/altconf/savinola.jpg", "assets/img/conference/altconf/searls.jpg", "assets/img/conference/altconf/segiddins.jpg", "assets/img/conference/altconf/simjp.jpg", "assets/img/conference/altconf/slaunchaman.jpg", "assets/img/conference/altconf/spacekatgal.jpg", "assets/img/conference/altconf/steipete.jpg", "assets/img/conference/altconf/taylanpince.jpg", "assets/img/conference/altconf/toddkerpelman.jpg", "assets/img/conference/altconf/tooluser.jpg", "assets/img/conference/altconf/twitter.png", "assets/img/conference/altconf/ultrasaurus.jpg", "assets/img/conference/altconf/uribaghin.jpg", "assets/img/conference/altconf/willowtreeapps.jpg", "assets/img/conference/altconf/wilshipley.jpg", "assets/img/conference/altconf/zats.jpg", "assets/img/docs/browser.png", "assets/img/docs/deployment/add-your-ssh-keys.png", "assets/img/docs/deployment/choose-a-datacenter.png", "assets/img/docs/deployment/choose-a-size.png", "assets/img/docs/deployment/choose-an-image.png", "assets/img/docs/deployment/distribution-ubuntu.png", "assets/img/docs/deployment/droplets.png", "assets/img/docs/deployment/finalize-and-create.png", "assets/img/docs/deployment/object-server-setup.png", "assets/img/docs/eclipse_annotation_processing.png", "assets/img/docs/eclipse_factory_path.png", "assets/img/docs/eclipse_java_devel_tools.png", "assets/img/docs/java/application-level-build-gradle.png", "assets/img/docs/java/project-level-build-gradle.png", "assets/img/docs/java/realmobject-watch-panel.png", "assets/img/docs/plugin.png", "assets/img/docs/react-native/realmreact-screenshot.png", "assets/img/docs/ros-rmp/Android-Studio.png", "assets/img/docs/ros-rmp/auth_cloudkit_addKey.png", "assets/img/docs/ros-rmp/BuildRealmTasks-for-Device.png", "assets/img/docs/ros-rmp/open-realmtasks-android.png", "assets/img/docs/ros-rmp/Realm-Dashboard-main.png", "assets/img/docs/ros-rmp/Realm-Dashboard-registration.png", "assets/img/docs/ros-rmp/Realm-Dashboard-Terminal.png", "assets/img/docs/ros-rmp/Realm-Mobile-Platform-DirList.png", "assets/img/docs/ros-rmp/Realm-Object-Server-macOS-Startup-KeyOutline.png", "assets/img/docs/ros-rmp/Realm-Object-Server-macOS-Startup.png", "assets/img/docs/ros-rmp/RealmBrowser-Conect-To-Object-Server.png", "assets/img/docs/ros-rmp/RealmBrowser-Initial-dialog.png", "assets/img/docs/ros-rmp/RealmBrowser.png", "assets/img/docs/ros-rmp/RealmTasks-Android-FirstTasks.png", "assets/img/docs/ros-rmp/RealmTasks-iOS-FirstTasks.png", "assets/img/docs/ros-rmp/realmTasks-list-trusted-certificate-.png", "assets/img/docs/ros-rmp/RealmTasks-macOS-FirstTasks.png", "assets/img/docs/ros-rmp/RealmTasks_Shared.png", "assets/img/docs/table-diagram.png", "assets/img/docs/tutorials/scanner/scanner-architecture.png", "assets/img/docs/tutorials/scanner/scanner-tutorial-step-1.png", "assets/img/docs/tutorials/scanner/scanner-tutorial-step-2.png", "assets/img/docs/tutorials/scanner/scanner-tutorial-step-3.png", "assets/img/docs/tutorials/scanner/scanner-tutorial-step-4.png", "assets/img/docs/tutorials/scanner/TexScannerTutorial-CameraL-and-Photos-Keys.png", "assets/img/docs/tutorials/scanner/TexScannerTutorial-EnableKeychainSharingCapability.png", "assets/img/favicons/favicon-114x114.png", "assets/img/favicons/favicon-120x120.png", "assets/img/favicons/favicon-144x144.png", "assets/img/favicons/favicon-152x152.png", "assets/img/favicons/favicon-16x16.png", "assets/img/favicons/favicon-32x32.png", "assets/img/favicons/favicon-57x57.png", "assets/img/favicons/favicon-72x72.png", "assets/img/favicons/favicon.ico", "assets/img/favicons/favicon.png", "assets/img/general/404.gif", "assets/img/general/fallbacks/post.jpg", "assets/img/general/swiftLogo.png", "assets/img/home/logos/3MGrey.png", "assets/img/home/logos/addidas.svg", "assets/img/home/logos/adidasGrey.png", "assets/img/home/logos/alibaba.svg", "assets/img/home/logos/alibabaGrey.png", "assets/img/home/logos/amazon.svg", "assets/img/home/logos/amazonGrey.png", "assets/img/home/logos/atlassian.svg", "assets/img/home/logos/avis.svg", "assets/img/home/logos/avisGrey.png", "assets/img/home/logos/awaGrey.png", "assets/img/home/logos/bandai.svg", "assets/img/home/logos/bandaiGrey.png", "assets/img/home/logos/bbc.svg", "assets/im
```

mrh-is commented:
Two things:

Try this in `config.rb`:

```ruby
    blog.custom_collections = {
      section: {
        template: "/news/section.html",
        link: "#{country_code unless country_code == :en}/news/sections/{section}.html",
        # We're ignoring the `property`, since we already know what it is
        transform: Proc.new { |article, property| article.data[:section][:name] }
      }
    }
```

The crazy first filename indicates that `article.data["section"]["name"]` isn't the right way to get the data. I'm trying symbols here, but you might have to play around with it.

Also you need to create `cn/news/section.html`. That error code seems to say that that template file doesn't exist.
